zerver/lib/digest.py

```diff
@@ -198,7 +198,7 @@ def gather_new_streams(user_profile: UserProfile,
         new_streams = list(get_active_streams(user_profile.realm).filter(
             is_web_public=True, date_created__gt=threshold))
 
-    elif user_profile.can_access_public_streams():
+    else:
```

Comment:
Looking at 867428719245e35bb4493295bed5282d50a9e1ab, I think I'd be happier with making the `is_guest` clause be the one that's an `else` clause, since that avoids the potential Zephyr logic regression (even if it'll never happen) and is more readable.  Especially since we may end up chaining a `can_access_web_public_streams` conditional in the future.

Reply:
Yep, makes sense.